app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt

```diff
@@ -112,13 +112,23 @@ class MessageActivity : AppCompatActivity() {
         }
 
         previewButton.setOnClickListener {
-            previewBadge.setValue(
-                    presenter.convertToPreview(if (!content.text.isEmpty()) content.text.toString() else " "),
-                    marquee.isChecked,
-                    flash.isChecked,
-                    Speed.values()[speed.selectedItemPosition],
-                    Mode.values()[mode.selectedItemPosition]
-            )
+            if (!content.text.isEmpty()) {
+                previewBadge.setValue(
+                        presenter.convertToPreview(content.text.toString()),
+                        marquee.isChecked,
+                        flash.isChecked,
+                        Speed.values()[speed.selectedItemPosition],
+                        Mode.values()[mode.selectedItemPosition]
+                )
+            } else {
+                previewBadge.setValue(
+                        Converters.convertDrawableToLEDHex((drawableRecyclerAdapter.getDefaultItem()).image) as java.util.ArrayList<String>,
```

Comment:
No casting

Reply:
@iamareebjamal I removed the redundant cast. Without this cast, there is an error.

Reply:
This cast is needed.

Reply:
Why is it needed? What's the type of the object without casting?

Reply:
Without casting, it is a List<String> while an ArrayList<String> is needed.

Reply:
Return List from everywhere, never use implementations, always use abstractions.

Reply:
Sorry, I did not get this. A list is being returned from `convertDrawableToLEDHex`. In order to make it a bitmap to be displayed on the preview, it then has to be converted into an ArrayList.

Reply:
Wherever the function is returning `ArrayList` change it to `List` 

Whichever function is accepting `ArrayList` as argument, change it to `List`

There's a reason why List is an interface and ArrayList implements it. There are 3-4 List implementations. Why use the concrete type of ArrayList unless you are depending on its specific implementation which we are not. 

https://stackoverflow.com/questions/13552413/depend-on-abstractions-do-not-depend-on-concrete-classes

Reply:
Ok. Got it. Thanks for sharing this information.

Reply:
Made the changes. Please check if it is alright.